sourcegraph/learn
# Move document fetching to lib

### What should this PR do?
Resolves [DEVED-82](https://sourcegraph.atlassian.net/browse/DEVED-82) by moving document fetching and parsing methods to the `lib` folder.

Rationale: Scoping the `util` folder to have only shared utility functions will ultimately make the project easier to reason about for newcomers; moving data fetching to `lib` also follows the conventions that Vercel lays out for analogous [projects](https://github.com/vercel/next.js/tree/canary/examples/blog-starter).

### Why are we making this change?
- We want to have things organized and scoped in such a way that folks can onboard to the project as quickly as possible.
- We want to follow organizational conventions that are recognized in Next.js projects to facilitate the above goal.

### What are the acceptance criteria? 
- Everything should work as it currently does in prod, and the organizational changes make sense.

### How should this PR be tested?
- Check out the branch, ensure everything works as it currently does in pod & sanity check organizational changes.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
